src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildRegistrationStartup.cs

```diff
@@ -31,6 +31,11 @@ public MSBuildRegistrationStartup(ILogger<MSBuildRegistrationStartup> logger)
             _logger = logger;
         }
 
+        public static string GetMSBuildPath()
```

Comment:
I don't care for the public static method on an instance class. Can we tweak this to either:

1. Be an instance method and register MSBuildRegistrationStartup as a singleton
2. Leave this service as transient but give it a dependency on MSBuildPathLocator and have it set the MSBuildPath property when it runs so that future services can get the property from (the singleton instance of) MSBuildPathLocator

Reply:
went with option 2 I liked the idea that there's only one place to get this value.